transactor/src/test/scala/io/mediachain/copycat/SeedingCopycatClient.scala

```diff
@@ -0,0 +1,97 @@
+package io.mediachain.copycat
+
+import cats.data.Xor
+import io.mediachain.copycat.Client.{ClientState, ClientStateListener}
+import io.mediachain.copycat.Dummies.DummyReference
+import io.mediachain.protocol.Datastore.{CanonicalRecord, ChainCell, ChainEntry, _}
+import io.mediachain.protocol.Transactor.{JournalCommitError, JournalError, JournalListener}
+
+import scala.concurrent.{ExecutionContext, Future, Promise}
+
+class SeedingCopycatClient(serverAddress: String)
+  (implicit executionContext: ExecutionContext)
+  extends JournalListener {
+
+  val client = Client.build()
+
+  def replayEntry(journalEntry: JournalEntry, client: Client, datastore: Datastore)
+    (implicit executionContext: ExecutionContext)
+  : Future[Xor[JournalError, JournalEntry]] = {
+    journalEntry match {
+      case e: CanonicalEntry => {
+        val canonical =
+          datastore.getAs[CanonicalRecord](e.ref)
+
+        canonical.map(c => client.insert(c)
+            .map { res =>
+              println(s"replayed entry ${journalEntry.index}. result: $res")
+              res
+            }
+        )
+          .getOrElse(
+            Future.failed(
+              new RuntimeException("Unable to replay journal entry: " +
+                  s"can't retrieve canonical for ref ${e.ref}")))
+      }
+      case e: ChainEntry => {
+        val cell = datastore.getAs[ChainCell](e.chain)
+
+        cell.map(c => client.update(c.ref, c)
+          .map { res =>
+            println(s"replayed entry ${journalEntry.index}. result: $res")
+            res
+        })
+          .getOrElse(
+            Future.failed(
+              new RuntimeException("Unable to replay journal entry: " +
+                  s"can't retrieve chain cell for ref ${e.ref}")))
+      }
+    }
+  }
+
+
+  def seed(blockchain: List[JournalBlock], datastore: Datastore): Future[Xor[JournalError, JournalEntry]] = {
+
+    val journalEntries = blockchain.flatMap(_.entries.toList).sortBy(_.index)
+
+    def empty: Future[Xor[JournalError, JournalEntry]] =
+      Future(Xor.right(CanonicalEntry(index = -1, ref = new DummyReference(-1))))
+
+    def serialize(f1: => Future[Xor[JournalError, JournalEntry]], f2: => Future[Xor[JournalError, JournalEntry]]) =
```

Comment:
well... the idea is to flatMap `f2` onto `f1` so that it executes after `f1` completes.  Since `Future.sequence` (which was my first thought) executes all futures in parallel.  In this case it bails out if the result of `f1` is an `Xor.Left`.